### PR TITLE
Backport #10883 and #10918

### PR DESCRIPTION
**Pull request by nicu-da**
Backport:
- https://github.com/digital-asset/daml/pull/10883 - this is an important test scenario for command deduplication (cherry picked from commit f4adee91ca182c9bac1847585491adbac7ef82dd)
- https://github.com/digital-asset/daml/pull/10918 - a fix to the test infra

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by hubert-da** [comment]
Requires https://github.com/digital-asset/daml/pull/10928

**Review by fabiotudone-da — APPROVED** [review]

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala [hunk 1]
```diff
@@ -16,76 +16,145 @@ import com.daml.ledger.api.testtool.infrastructure.Assertions.fail
...
+      val alreadyExistsResponses = responses.getOrElse(Code.ALREADY_EXISTS, 0)
+      // Participant-based command de-duplication can currently also reject duplicates via a SQL exception when using the CommandSubmissionService
+      val internalResponses = responses.getOrElse(Code.INTERNAL, 0)
+      // Canton can return ABORTED for duplicate submissions
+      // Participant based command de-duplication can currently also return ABORTED when using the CommandService
+      val abortedResponses = responses.getOrElse(Code.ABORTED, 0)
+      val duplicateResponses =
+        alreadyExistsResponses + internalResponses + abortedResponses
+      assert(
+        okResponses == 1 && duplicateResponses == numberOfParallelRequests - 1,
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala:108]:
```suggestion
        okResponses == 1 && duplicateResponses == expectedDuplicateResponses,
```

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala [hunk 2]
```diff
@@ -16,76 +16,145 @@ import com.daml.ledger.api.testtool.infrastructure.Assertions.fail
...
+      request,
+      submitAndWaitRequestAndGetStatusCode(ledger)(_, party),
+    )
+  })
+
+  private def runTestWithSubmission[T](
+      ledger: ParticipantTestContext,
+      party: Party,
+      request: T,
+      submitRequestAndGetStatus: T => Future[Code],
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala:84]:
```suggestion
      submitRequestAndGetStatusCode: T => Future[Code],
```

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala [hunk 3]
```diff
@@ -16,76 +16,145 @@ import com.daml.ledger.api.testtool.infrastructure.Assertions.fail
...
 import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.v1.command_service.SubmitAndWaitRequest
 import com.daml.ledger.api.v1.command_submission_service.SubmitRequest
 import com.daml.ledger.api.v1.commands.Commands.DeduplicationPeriod
+import com.daml.ledger.client.binding.Primitive
 import com.daml.ledger.client.binding.Primitive.Party
-import com.daml.ledger.test.model.Test.DummyWithAnnotation
+import com.daml.ledger.test.model.Test.{Dummy, DummyWithAnnotation}
 import io.grpc.Status
 import io.grpc.Status.Code
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala:26]: I prefer not to import `Code` and instead use `Status.Code`. I find it clearer. Your call though.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala [hunk 4]
```diff
@@ -16,76 +16,145 @@ import com.daml.ledger.api.testtool.infrastructure.Assertions.fail
...
+      ec: ExecutionContext
+  ) = {
+    submitResult
+      .transformWith {
+        case Failure(exception) =>
+          exception match {
+            case GrpcException(status, _) =>
+              Future.successful(status.getCode)
+            case NonFatal(otherException) =>
+              fail(s"Not a GRPC exception $otherException", otherException)
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala:145]: Why include the exception twice?

```suggestion
              fail(s"Not a GRPC exception.", otherException)
```

**Review by SamirTalwar-DA — APPROVED** [review]

**Comment by SamirTalwar-DA** [comment]
None of my comments are mandatory for the backport. I just saw them now and I figured we can address these on `main`, so I wanted to capture them.

**Pull request merged**